nibabies/utils/bids.py

```diff
@@ -304,3 +304,79 @@ def validate_input_dir(exec_env, bids_dir, participant_label):
             subprocess.check_call(["bids-validator", str(bids_dir), "-c", temp.name])
         except FileNotFoundError:
             print("bids-validator does not appear to be installed", file=sys.stderr)
+
+
+def generate_bids_skeleton(location, _bids_dict):
+    """
+    Converts a BIDS directory in dictionary form to a file structure.
+    """
+    from copy import deepcopy
+
+    # ensure dictionary remains unaltered
+    bids_dict = _bids_dict.copy()
```

Comment:
Private variables as parameters don't really make sense. Also, I think you want `deepcopy`:

```Python
In [1]: from copy import deepcopy

In [2]: a = {"a": ["b"], "c": ["d"]}

In [3]: b = a.copy()

In [4]: c = deepcopy(a)

In [5]: b["a"].append("e")

In [6]: c["a"].append("f")

In [7]: a
Out[7]: {'a': ['b', 'e'], 'c': ['d']}

In [8]: b
Out[8]: {'a': ['b', 'e'], 'c': ['d']}

In [9]: c
Out[9]: {'a': ['b', 'f'], 'c': ['d']}
```

```suggestion
def generate_bids_skeleton(location, bids_dict):
    """
    Converts a BIDS directory in dictionary form to a file structure.
    """
    from copy import deepcopy

    # ensure dictionary remains unaltered
    bids_dict = deepcopy(bids_dict)
```